src/events/EventTileFactory.tsx

```diff
@@ -211,9 +214,17 @@ export function pickFactory(mxEvent: MatrixEvent, cli: MatrixClient, asHiddenEv?
 
     // Try and pick a state event factory, if we can.
     if (mxEvent.isState()) {
+        if (
+            M_BEACON_INFO.matches(evType) &&
+            SettingsStore.getValue("feature_location_share_live")
+        ) {
+            return STATE_EVENT_TILE_TYPES[M_BEACON_INFO.name];
```

Comment:
can we instead do something similar to Jitsi where the event type isn't part of the map but instead overrides the factory at the last possible second? Means fewer accidental renders if things slip by.